docs/build-css-js.md

```diff
@@ -0,0 +1,51 @@
+# Building CSS and JS
+
+This example uses `npm` and [Webpack Encore](https://symfony.com/doc/current/frontend.html).
+
+Modify `Dockerfile` to build your assets before anything else.
+The output in `public/build` (the configuration default) will then be copied to both caddy and php.
+
+In your production environment, this is all you need for an immutable release.
+
+```diff
+ARG CADDY_VERSION=2
++ ARG NODE_VERSION=17
+
++ FROM node:${NODE_VERSION}-alpine AS symfony_npm
++ WORKDIR /srv/app
++ COPY . .
++ RUN npm install; \
++     npm run build
+
+# "php" stage
+```
+
+```diff
+VOLUME /srv/app/var
+
++ COPY --from=symfony_npm /srv/app/public/build public/build
+
+ENTRYPOINT ["docker-entrypoint"]
+CMD ["php-fpm"]
+
+FROM caddy:${CADDY_VERSION}-builder-alpine AS symfony_caddy_builder
+```
+
+Modify `docker-compose.override.yml` to add an `npm` container in your development environment and watch for file changes.
+
+```diff
+  caddy:
+    volumes:
+      - ./docker/caddy/Caddyfile:/etc/caddy/Caddyfile:ro
+      - ./public:/srv/app/public:ro
+
++  npm:
++    build:
++      context: .
++      target: symfony_npm
++    depends_on:
++      - caddy
```

Comment:
```suggestion
```

To be double-checked, but AFAIU there is no dependency to the Caddy container here.

Reply:
Yes you are correct, I left that in by mistake.